docs/sources/alerting/unified-alerting/alerting-rules/create-grafana-managed-rule.md

```diff
@@ -112,6 +112,23 @@ The following template variables are available when expanding annotations and la
 | $values | The values of all reduce and math expressions that were evaluated for this alert rule. For example, `{{ $values.A }}`, `{{ $values.A.Labels }}` and `{{ $values.A.Value }}` where `A` is the `refID` of the expression. This is unavailable when the rule uses a classic condition. |
 | $value  | The value string of the alert instance. For example, `[ var='A' labels={instance=foo} value=10 ]`.                                                                                                                                                                                  |
 
+#### Template functions
+
+The following template functions are available when expanding annotations and labels.
+
+| Name               | Argument         | Return                 | Description                                                                                                                        |
+| ------------------ | ---------------- | ---------------------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| humanize           | number or string | string                 | Converts a number to a more readable format, using metric prefixes.                                                                |
```

Comment:
I think we're missing a few from this list - even if we don't support them yet, I think we should list them. Can we make sure we've tested their behaviour to make sure they don't panic or something similar?

From what I can see:

- `query`
- `first`
- `label`
- `value`
- `strvalue`
- `reReplaceAll`
- `safeHtml`
- `graphLink`
- `tableLink`
- `sortByLabel`
- `pathPrefix`
- `externalURL`

Are not mentioned here - the reason why this is important is that Prometheus is coming up together with a conformance program for alert generation and all of these functions will be tested as part of the it - for each that we don't support that lowers our compliance score.

Reply:
They aren't listed because we are passing a no-op as the `QueryFn` when we create a `NewTemplateExpander()`. I'll make sure we list them and have tests for those functions too!

Reply:
- `query` , `first` , `label` , `value` , `sortByLabel` and `strvalue` return an empty string.
- `reReplaceAll` wasn't documented for some reason... I just solved that and wrote a test case in a new commit.
- `safeHtml` is not enabled since we're using `TemplateExpander.Expand()`, not `TemplateExpander.ExpandHTML()` , and HTML is being escaped by default. Prometheus uses `ExpandHTML()` for HTML files.
- `graphLink` and `tableLink` work as expected, but we have to check with FE team whether or not the links provided by Prometheus' functions match our links for graphs and tables.
- `pathPrefix` and `externalURL` return an empty string since we're using `nil` as the value for `*url.URL` in `NewTemplateExpander()` .

Btw, none of the functions panics.

Reply:
> pathPrefix and externalURL

I think we can fix those, grafana has a setting that exposes the external URL of the instance - I think we should use those.

> graphLink and tableLink work as expected, but we have to check with FE team whether or not the links provided by Prometheus' functions match our links for graphs and tables.

Then, let's align with them and check to make sure they work as expected.